class/practice_3.1/Helpers.cpp

```diff
@@ -0,0 +1,20 @@
+#include "Helpers.hpp"
+#include <algorithm>
+#include <string>
+
+double dummyFunc(double x) {
+    return x > 0 ? x : 0;
+}
+
+std::string repeatString(const std::vector<int>& numbers, const std::string& str) {
+    if (numbers.empty()) {
```

Comment:
It may be helpful for you to read about std::views::repeat, std::views::join, std::views::take that function together with std::ranges::max can simplify this function a lot to a couple of lines.
For example, max number is just a one-liner `auto repeatCount = numbers.empty() ? 0 : std::ranges::max(numbers);`